### PR TITLE
[template/angular-xmcloud] Fix SXA component styles for basic site

### PR DESCRIPTION
**Pull request by sc-addypathania**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
- Fix styles issues caused by extra Angular selector tags in DOM tree.
- Fix double inclusion of bootstrap styles in bundle.

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/promo/promo.component.ts [hunk 1]
```diff
@@ -4,5 +4,22 @@ import { SxaComponent } from '../sxa.component';
 @Component({
   selector: 'app-promo',
   templateUrl: './promo.component.html',
+  styles: [
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/promo/promo.component.ts:7]: Can we create a separate SCSS file to store the styles?

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts [hunk 2]
```diff
@@ -1,16 +1,55 @@
-﻿import { Component, OnInit } from '@angular/core';
+﻿import { Component, OnInit, ElementRef, Renderer2 } from '@angular/core';
 import { Field } from '@sitecore-jss/sitecore-jss-angular';
 import { SxaComponent } from '../sxa.component';
 
 @Component({
   selector: 'app-richtext',
   templateUrl: './richtext.component.html',
+  styles: [
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts:8]: Same as above regarding SCSS file

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts [hunk 3]
```diff
@@ -1,16 +1,55 @@
...
+    `
+      :host(.footer-richtext) {
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts:10]: Inside all of these styles I see a lot of "magic" numbers
e.g. "width: 17%", "nth-child(2)", "max-width: 992px"
to it means that apply kind of "hack" to this component, so the solution is not reliable

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts [hunk 4]
```diff
@@ -1,16 +1,55 @@
...
+  }
+
   ngOnInit() {
     super.ngOnInit();
     this.text = this.rendering.fields?.Text as Field<string>;
+
+    let parent = this.el.nativeElement.parentElement;
+
+    while (parent) {
+      if (parent.id === 'footer') {
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts:48]: We can't use such condition and rely on specific parent component
it's a "hack", what will happen if id of the parent element is changed but it's actually the same "footer" placeholder?

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts (hunk 3 above)
Comment (yavorsk) [reply packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts:10]: maybe we can explore using attribute selectors for components instead element selectors?

**Pull request closed without merging**